<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <!-- ICP params -->
    <arg name="num_iterations"          default="1000000"/>
    <arg name="max_dist"                default="0.5"/>
    <arg name="translation_epsilon"     default="0.0000000000001"/>
    <arg name="model_fit_epsilon"       default="0.0000000000001"/>

    <!-- Template Matching Params -->
    <arg name="match_frame"             default="base_link"/>
    <arg name="cloud_topic"             default="/head_camera/depth_registered/points"/>
    <arg name="template_filename"       default="bin.pcd"/>
    <arg name="initial_estimate"        default="0 0 0 0 0 0"/>
    <arg name="template_offset"         default="0 0 0 0 0 0"/>
    <arg name="output_frame"            default="template_pose"/>
    <arg name="visualize_output"        default="true"/>
    <arg name="debug"                   default="true"/>
    <arg name="latch_initial_estimate"  default="false"/>
    <arg name="provide_processed_cloud" default="true"/>

    <!-- launch template matching for demo world with schunk machine -->
    <node pkg="fetchit_icp" type="template_matcher_node" name="kit_template_matcher_node" output="screen">
        <param name="matching_frame"          value="$(arg match_frame)"/>
        <param name="pcl_topic"               value="$(arg cloud_topic)"/>
        <param name="template_file"           value="$(arg template_filename)"/>
        <param name="initial_estimate_string" value="$(arg initial_estimate)"/>
        <param name="template_offset_string"  value="$(arg template_offset)"/>
        <param name="template_frame"          value="$(arg output_frame)"/>
        <param name="visualize"               value="$(arg visualize_output)"/>
        <param name="debug"                   value="$(arg debug)" />
        <param name="latch_initial"           value="$(arg latch_initial_estimate)"/>
        <param name="pre_processed_cloud"     value="$(arg provide_processed_cloud)"/>
    </node>

    <!-- assumes this launched by schunk detector
    <include file="$(find fetchit_icp)/launch/start_icp_matching.launch">
        <arg name="num_iterations"      value="$(arg num_iterations)"/>
        <arg name="max_dist"            value="$(arg max_dist)"/>
        <arg name="translation_epsilon" value="$(arg translation_epsilon)"/>
        <arg name="model_fit_epsilon"   value="$(arg model_fit_epsilon)"/>
    </include>-->

</launch>
